apiserver/facades/controller/firewaller/firewaller_test.go

```diff
@@ -47,7 +51,9 @@ func (s *firewallerSuite) SetUpTest(c *gc.C) {
 		cloudspec.MakeCloudSpecCredentialContentWatcherForModel(s.State),
 		common.AuthFuncForTag(s.Model.ModelTag()),
 	)
-	controllerConfigAPI := common.NewStateControllerConfig(s.State)
+
+	s.ctrl = gomock.NewController(c)
+	controllerConfigAPI := mocks.NewMockControllerConfigAPI(s.ctrl)
```

Comment:
I don't think we should create s.ctrl  in SetUpTest.
But it's ok here because there are no assertions for controllerConfigAPI at all.
So nothing to check in s.ctrl.Finish().

Reply:
We need to do it because of how the test suites have been set up and how the facade operates. Previously, the controller config component that needed the mock controller was optional and could be added later. Now we need it at the time of set up and there are embedded test fixtures that fail unless set up is done here.